vshard/storage/init.lua

```diff
@@ -1091,17 +1091,26 @@ local function vclock_lesseq(vc1, vc2)
 end
 
 local function wait_lsn(timeout, interval)
+    local current_id = box.info().id
     local vclock = box.info.vclock
     local deadline = fiber_clock() + timeout
```

Comment:
Nice fix, thanks for figuring this out! I doubt it is caused by any of our commits. Something must have changed in core, because this function didn't change for ages except a recent patch when the function was basically just renamed.
I would advice to consider this diff though, in order not to call the expensive `box.info` too many:
```Diff
 local function wait_lsn(timeout, interval)
-    local current_id = box.info().id
-    local vclock = box.info.vclock
+    local info = box.info
+    local current_id = info.id
+    local vclock = info.vclock
     local deadline = fiber_clock() + timeout
```

Reply:
Sounds reasonable. Fixed